### ISSUE TITLE
Still more things to mark obsolete in 1.x that are removed in 2.0

### ISSUE THREAD

**Issue by airbreather** [NetTopologySuite/NetTopologySuite#318]
I thought the changes on #316 were going to be odd ones out, but the more I inspect the public API after some of the bigger changes, the more things I find that I either missed or simply didn't think of.

I'm going to keep this open for a while longer so it can be resolved with more-or-less one more "mark deprecated" commit in v1 + one more round in v2 to remove it.

- [ ] Interfaces removed per NetTopologySuite/GeoAPI#68.
- [x] `Geometry.Equals(Geometry)` is [marked](https://github.com/locationtech/jts/blob/d1382a0ba7c4d7d00789f6a6cd9157ffb873aaf0/modules/core/src/main/java/org/locationtech/jts/geom/Geometry.java#L1031) as being there for "backward compatibilty reasons" in JTS.~~
- [ ] `GeometryEx` static class (in `Geometry.cs`)
- [x] ~~Certain constructors for geometry types are deprecated in JTS but not in NTS (the ones that don't accept an instance of `GeometryFactory`)~~

**Comment by airbreather** [comment]
> 2\. `Geometry.Equals(Geometry)` is [marked](https://github.com/locationtech/jts/blob/d1382a0ba7c4d7d00789f6a6cd9157ffb873aaf0/modules/core/src/main/java/org/locationtech/jts/geom/Geometry.java#L1031) as being there for "backward compatibilty reasons" in JTS.

If nothing else, this one is probably going to demand a special call-out in the documentation, since code like this will still compile, but the semantics will change:
```csharp
Geometry g1 = GetGeom1();
Geometry g2 = GetGeom2();
bool eq = g1.Equals(g2); // v1: topological equality, v2: structural equality
```

**Comment by FObermaier** [comment]
I don't think that removing `Equals(Geometry)` from the public api is a good idea.
While having it there does not do any harm it puts additional effort on porting code.

**Comment by airbreather** [comment]
> I don't think that removing `Equals(Geometry)` from the public api is a good idea.

Reverted in 2296109.

> While having it there does not do any harm

FWIW, I disagree with this one.  Equality in .NET is complicated enough on types that **don't** have two methods called `.Equals`, each with purposely different semantics.

Half the XML documentation for `Equals(Geometry)` and a sentence or two of the documentation for the `Geometry` type are also dedicated to a combination of warning people about potential for using it incorrectly and pointing to an alternative that's more explicit.

**Comment by airbreather** [comment]
> 4\. Certain constructors for geometry types are deprecated in JTS but not in NTS (the ones that don't accept an instance of `GeometryFactory`)

Perhaps all constructors should be made `protected`... anyway, this part isn't going to happen in 2.0, so pushing that to a different milestone.

**Comment by airbreather** [comment]
1.x is done. Closing.
